web/modules/custom/asset_release/src/Controller/AssetReleaseController.php

```diff
@@ -73,18 +74,29 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetReleaseAccess(RdfInterface $rdf_entity) {
-    // Check that the passed in RDF entity is a collection, and that the user
-    // has the permission to create asset_releases.
-    // @todo Collection owners and facilitators should also have the right to
-    //   create asset_releases for the collections they manage.
-    // @see https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-2448
-    if ($rdf_entity->bundle() == 'solution' && $this->currentUser()
-        ->hasPermission('create asset_release rdf entity')
-    ) {
-      return AccessResult::allowed();
+    $user = \Drupal::currentUser();
```

Comment:
Don't access *any* of the static methods that are in the `Drupal` class from inside OO code. That class is purely intended as an interface between procedural code and OO code. It makes no sense to bridge from OO -> procedural -> OO.

You can directly inject the `AccountProxy` service, to get the current user object from it.